orchagent/pfcwdorch.cpp

```diff
@@ -400,24 +401,33 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::registerInWdDb(const Port& port,
         return;
     }
 
+    set<uint8_t> losslessTc;
+    uint8_t pfcMask = attr.value.u8;
+    for (uint8_t i = 0; i < PFC_WD_TC_MAX; i++)
+    {
+        if ((pfcMask & (1 << i)) == 0)
+        {
+            continue;
+        }
+
+        losslessTc.insert(i);
+    }
+
     if (!c_portStatIds.empty())
     {
         string key = getFlexCounterTableKey(sai_serialize_object_id(port.m_port_id));
         vector<FieldValueTuple> fieldValues;
+        // Only register lossless tc counters in database.
         string str = counterIdsToStr(c_portStatIds, &sai_serialize_port_stat);
-        fieldValues.emplace_back(PORT_COUNTER_ID_LIST, str);
+        string losslessStr = getLosslessStr(str,losslessTc);
+        fieldValues.emplace_back(PORT_COUNTER_ID_LIST, losslessStr);
 
         m_flexCounterTable->set(key, fieldValues);
     }
 
-    uint8_t pfcMask = attr.value.u8;
-    for (uint8_t i = 0; i < PFC_WD_TC_MAX; i++)
+    for (auto i : losslessTc)
     {
-        if ((pfcMask & (1 << i)) == 0)
-        {
-            continue;
-        }
-
+        SWSS_LOG_NOTICE("current tc is %d", i);
```

Comment:
this is more like debug message, not really an external visible event.

Reply:
even for debug message, you need to make sure people can understand the meaning of the message when they only look at the message and know nothing about the code.

image yourself know nothing about the code and see a message "current tc is 1" in the syslog. what can he get from the message.

---
In reply to: [176648249](https://github.com/Azure/sonic-swss/pull/455#discussion_r176648249) [](ancestors = 176648249)